pyomo/core/base/reference.py

```diff
@@ -348,62 +387,103 @@ def _get_iter(self, _slice, key):
 
 
 def _identify_wildcard_sets(iter_stack, index):
-    # if we have already decided that there isn't a comon index for the
+    # if we have already decided that there isn't a common index for the
     # slices, there is nothing more we can do.  Bail.
     if index is None:
         return index
 
     # Walk the iter_stack that led to the current item and try to
     # identify the component wildcard sets
+    #
+    # `tmp` will be a list of dicts mirroring the iter_stack. Each
+    # dict maps position within that level's component's "subsets"
+    # list to the set at that position if it is a wildcard set.
     tmp = [None]*len(iter_stack)
     for i, level in enumerate(iter_stack):
         if level is not None:
             offset = 0
+            # `offset` is the position, within the total index tuple,
+            # of the first coordinate of a set.
+            # This function seems to only make sense if 
+            # normalize_index.flatten is True.
             wildcard_sets = {}
+            # `wildcard_sets` maps position in the "subsets list"
+            # to its set if that set is a wildcard?
             for j,s in enumerate(level.component.index_set().subsets()):
+                # Iterate over the sets that could possibly be wildcards
                 if s is UnindexedComponent_set:
+                    # Why is this treated as a wildcard? (How is this used?)
```

Comment:
You can hit this weirdness when scalars are treated as indexed components (explicitly or implicitly):
```python
m = ConcreteModel()
m.x = Var()
m.r1 = Reference(m.x)
m.r2 = Reference(m.x[...])

m.b = Block()
m.b.y = Var()
m.r3 = Reference(m.b[:].m.y[:])
```